main/systemd.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+    "fmt"
+    "os"
+    "os/exec"
+    "strings"
+    "path/filepath"
+)
+
+func contains(distros []string, distro string) bool {
+    if distro != "" {
+        for _, v := range distros {
+            if v == distro {
+                return true
+            }
+        }
+    }
+    return false
+}
+
+func get_distro() string {
+    path, _ := os.Getwd()
+    output, err := exec.Command("sudo", filepath.Join(path, "scripts/distro_script.sh")).Output()
+    if err != nil {
+        errorMessage := fmt.Sprintf("Error getting distro name: %+v", err)
+		errorLogger.Println(errorMessage)
+    }
+    distroName := strings.ToLower(string(output))
+    infoLogger.Println(fmt.Sprintf("distro name: %s", distroName))
+    return distroName
+}
+
+func getSystemdUnitFileInstallPath() string {
+
+    var systemdPath string
+
+    distro := get_distro()
```

Comment:
what if this function faces an issue? you're not passing the errors as expected for golang. Please add that

Reply:
what sort of errors we expect from get_distro. Already checking inside function and logging appropriately.

Reply:
> what sort of errors we expect from get_distro

Right out of the bat, the first thing that comes to mind is what if your shell script fails with a non-0 exit code? The script is using a lot of different utilities based on some files but what if something is wrong there and it fails? What if we don;t have sudo privileges to run this command?

Plethora of things can go wrong, there's no harm in having a proper error handling in place. 

For your use case though, I don't mind returning the default `systemd`  path value incase we're unable to determine the distro, but that check should be inside this function rather than the `getDistro` function, the `getDistro` should still return an error incase it can't determine the distro. 

Here's a useful link I found about error handling in golang, might be a worthwhile read - https://gabrieltanner.org/blog/golang-error-handling-definitive-guide

Reply:
`Right out of the bat, the first thing that comes to mind is what if your shell script fails with a non-0 exit code? The script is using a lot of different utilities based on some files but what if something is wrong there and it fails? What if we don;t have sudo privileges to run this command?`

These errors you get from from stderr, you pass this to and catch at same place doesn't make a difference in this scenario. we still achieving same goal. I don't mind changing and following convention.

Reply:
> I don't mind changing and following convention

Functionally it doesn't make sense, but at the same time convention is what we need to keep the state of the repo clean, easy to read and make it futureproof. I would really appreciate it if you did follow it, thank you :)

Reply:
included error handling